core/Implement.carp

```diff
@@ -0,0 +1,8 @@
+(defndynamic implement-declaration [mod interface]
```

Comment:
Do you think these could live in the general `Macros.carp` or in `Interface.carp`? Since they're just helpers for implementing interfaces, seems odd to have their own file. But if this makes more sense from a --no_core perspective, it's fine!

Reply:
Heh, sure – maybe I went a little overboard with splitting up stuff :) I just felt that Macros.carp became such a mix of wildly different things, but maybe that's OK.